module/epochs/qc_client.go

```diff
@@ -75,34 +76,38 @@ func (c *QCContractClient) SubmitVote(ctx context.Context, vote *model.Vote) err
 	}
 
 	// get account details
-	account, err := c.client.GetAccount(ctx, sdk.BytesToAddress([]byte(c.accountAddress)))
+	account, err := c.client.GetAccount(ctx, c.accountAddress)
 	if err != nil {
 		return fmt.Errorf("could not get account: %w", err)
 	}
 
 	// sign transaction
-	sk, err := sdkcrypto.DecodePrivateKey(sdkcrypto.ECDSA_P256, c.privateKey.Encode())
-	signer := sdkcrypto.NewInMemorySigner(sk, sdkcrypto.SHA2_256)
+	sk, err := sdkcrypto.DecodePrivateKeyHex(account.Keys[c.accountKeyIndex].SigAlgo, c.privateKey)
```

Comment:
This can panic if the `accountKeyIndex` is out of bounds -- it would be better to handle this explicitly. 

Given that we can't change the configuration of this component at runtime anyway, we could probably get the account in the constructor and do this check there, as well as construct the `Signer` object. That will take some logic out of the main `Vote` method and allow us to fail more quickly with bad configuration.